src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,207 @@
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String insertRequest = "INSERT INTO cars (model, manufacturer_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createCarStatement = connection
+                        .prepareStatement(insertRequest, Statement.RETURN_GENERATED_KEYS)) {
+            createCarStatement.setString(1, car.getModel());
+            createCarStatement.setLong(2, car.getManufacturer().getId());
+            createCarStatement.executeUpdate();
+            ResultSet generateKeys = createCarStatement.getGeneratedKeys();
+            if (generateKeys.next()) {
+                Long id = generateKeys.getObject(1, Long.class);
+                car.setId(id);
+            }
+        } catch (SQLException throwable) {
+            throw new RuntimeException("Can't insert car to DB ", throwable);
+        }
+        addDriverToCar(car);
+        return car;
+    }
+
+    private void addDriverToCar(Car car) {
+        String insertDriversQuery = "INSERT INTO cars_drivers (car_id, driver_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement addDriverToCarStatement = connection
+                        .prepareStatement(insertDriversQuery)) {
+            addDriverToCarStatement.setLong(1, car.getId());
+            for (Driver driver: car.getDrivers()) {
+                addDriverToCarStatement.setLong(2, driver.getId());
+                addDriverToCarStatement.executeUpdate();
+            }
+        } catch (SQLException throwable) {
+            throw new RuntimeException("Can't add driver to car ", throwable);
+        }
+    }
+
+    private void removeDriverFromCar(Car car) {
+        String deleteDriver = "DELETE FROM cars_drivers WHERE car_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement removeDriverStatement = connection
+                         .prepareStatement(deleteDriver)) {
+            removeDriverStatement.setLong(1, car.getId());
+            removeDriverStatement.executeUpdate();
+        } catch (SQLException throwable) {
+            throw new RuntimeException("Can't delete driver from car: ", throwable);
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String selectRequest = "SELECT c.id AS car_id, model, m.id AS manufacturer_id, "
+                + "m.name, m.country FROM cars c JOIN manufacturers m "
+                + "ON c.manufacturer_id = m.id WHERE c.id = ? AND c.is_deleted = FALSE";
```

Comment:
```suggestion
        String selectRequest = "SELECT c.id AS car_id, model, m.id AS manufacturer_id, "
                + "m.name, m.country FROM cars c "
                + "JOIN manufacturers m ON c.manufacturer_id = m.id "
                + "WHERE c.id = ? AND c.is_deleted = FALSE";
```
It is better to move 'join' to the new line

Reply:
Ok